WordPress/gutenberg
# Framework: Upgrade ESLint dependencies

This pull request seeks to upgrade ESLint dependencies to their latest versions. This brings some improvements to capturing false negatives and in some cases stricter rules. Upgrading [eslint-config-wordpress](https://github.com/WordPress-Coding-Standards/eslint-config-wordpress) brings with it some additional rules, some of which are respected here ([multi-line ternaries](https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/#multi-line-statements)) and some disabled ([yoda](https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/#yoda-conditions) which, until recently, was optional; probably worth a chat about).

__Implementation notes:__

Most changes should be purely cosmetic, except:

- In token field, I removed a `ref` which was applied and used in tests. These tests fail already in master and may arguably be better off removed since they're skipped except when passing an environment variable. I did my best to restore basic functionality. As far as I can tell, the ref was not used anywhere except in these tests.

__Testing instructions:__

Verify that no errors or warnings are logged on lint:

```
npm run lint
```

Review code to verify all renamings are broadly applied, and no regressions occur in basic usage of the editor application.
